src/CoreShop/Component/Order/Cart/Rule/Condition/VoucherConditionChecker.php

```diff
@@ -46,13 +51,31 @@ public function isCartRuleValid(OrderInterface $cart, CartPriceRuleInterface $ca
             return false;
         }
 
+
         // max usage per code condition
         if (is_numeric($maxUsagePerCode)) {
             if ($maxUsagePerCode != 0 && $storedCode->getUses() >= $maxUsagePerCode) {
                 return false;
             }
         }
 
+        // max usage per user condtion
+        if (is_numeric($maxUsagePerUser)){
+
+            $user = $this->security->getUser();
```

Comment:
The reason we get the user from the security component, is that we wanted to explicitly limit the vouchers for registered customers only, because it's just not really possible for guests. This means if you use such a voucher as a guest, it will always be invalid.

Reply:
Ok, but should we still rename user to customer as in:
https://github.com/coreshop/CoreShop/pull/2451#discussion_r1402112269

and still pass the CustomerInterface in:
https://github.com/coreshop/CoreShop/pull/2451#discussion_r1402113025

if we only work with the CoreShopUser?

Reply:
In reality, it also works for Guest Customers. CoreShop re-uses a customer based on the email address. (that is a feature that if a customer wants to convert to a real user, we have the order history)